coalib/bearlib/abstractions/Linter.py

```diff
@@ -0,0 +1,680 @@
+from contextlib import contextmanager
+import inspect
+from itertools import compress
+import re
+import shutil
+from subprocess import check_call, CalledProcessError, DEVNULL
+from types import MappingProxyType
+
+from coalib.bears.LocalBear import LocalBear
+from coalib.misc.ContextManagers import make_temp
+from coalib.misc.Decorators import assert_right_type, enforce_signature
+from coalib.misc.Future import partialmethod
+from coalib.misc.Shell import run_shell_command
+from coalib.results.Diff import Diff
+from coalib.results.Result import Result
+from coalib.results.RESULT_SEVERITY import RESULT_SEVERITY
+from coalib.settings.FunctionMetadata import FunctionMetadata
+
+
+def _prepare_options(options):
+    """
+    Prepares options for ``Linter`` for a given options dict in-place.
+
+    :param options:
+        The options dict that contains user/developer inputs.
+    """
+    allowed_options = {"executable",
+                       "output_format",
+                       "use_stdin",
+                       "use_stdout",
+                       "use_stderr",
+                       "config_suffix",
+                       "prerequisite_check_command"}
+
+    if not options["use_stdout"] and not options["use_stderr"]:
+        raise ValueError("No output streams provided at all.")
+
+    if options["output_format"] == "corrected":
+        if "diff_severity" in options:
+            if options["diff_severity"] not in RESULT_SEVERITY.reverse:
```

Comment:
And btw it must be 
```
"diff_severity" in options and
options["diff_severity"] not in RESULT_SEVERITY.reverse)
```
otherwise the body can't access `diff_severity` if it's not inside the options dict :)